ir/src/constraint_builder/mod.rs

```diff
@@ -184,12 +183,10 @@ impl ConstraintBuilder {
     /// then saved in the appropriate constraint list (boundary, validity, or transition).
     fn insert_constraint(
         &mut self,
-        lhs: ExprDetails,
-        rhs: ExprDetails,
+        root: NodeIndex,
+        trace_segment: usize,
+        domain: ConstraintDomain,
     ) -> Result<(), SemanticError> {
```

Comment:
nit: comments for this function need to be updated - but maybe this point is muted in subsequent PRs?

Reply:
I noted this generally below, but I plan to do a pass to update all comments and add all new ones after all of these refactors are merged. This way it won't be blocking and they'll only need to be updated once.